bigchaindb/pipelines/block.py

```diff
@@ -55,19 +55,20 @@ def validate_tx(self, tx):
             tx (dict): the transaction to validate.
 
         Returns:
-            The transaction if valid, ``None`` otherwise.
+            The transaction (Transaction) if valid, ``None`` otherwise.
         """
-        if self.bigchain.transaction_exists(tx['id']):
+        tx = Transaction.from_dict(tx)
+        if self.bigchain.transaction_exists(tx.id):
```

Comment:
I'm not sure.  I think the established validation returns `False` if the transaction is in an undecided or invalid block, where the `block` process here treats undecided and valid blocks as the same.